### PR TITLE
Add rest-server jar

### PR DESCRIPTION
**Pull request by Ingvord**
This PR adds minimal working configuration of Tango REST server th SourceDistribution. 

Usage after `./configure --prefix=<install-prefix> && make && make install`:

`<intall-prefix>/bin/TangoRestServer rest` -- startup script
`<intall-prefix>/bin/tango_wrest` -- wrapper script script

* to build the distribution clone/download this PR's branch and run `ant build package && cd build/distr`

TangoRestServer is registered in TangoDb as sys/rest/0 with default configuration (no properties are defined)

This PR depends on changes in TangoDatabase, namely: [PR#12](https://github.com/tango-controls/TangoDatabase/pull/12)

To do:
- [x] add reset-server jar
- [x] add TangoRestServer to TangoDatabase: sys/rest/0
- [x] put TangoRestServer script next to DatabaseDs etc
- [x] add TangoRestServer to startup scripts (tango, tango_wca)

### CODE REVIEW

File: distribution.properties [hunk 1]
```diff
@@ -15,10 +16,11 @@ dbbench-ver=1.3
 jssh-terminal-ver=1.11
 log-viewer-ver=2.0.5
 pogo-ver=9.6.16
+rest-server-ver=1.2
 #tool_panels=
 #cppserver
 TangoTest=TangoTest-Release-2.1
-TangoDatabase=DataBase-Release-5.6
+TangoDatabase=Ingvord-add-rest-server
```
Comment (bourtemb) [inline distribution.properties]: Of course, this line will need to be changed when https://github.com/tango-controls/TangoDatabase/pull/12 will be merged in.

**Review by bourtemb — COMMENTED** [review]
On an Ubuntu 18.04LTS virgin VM, I tried to generate the Tango Source distribution package (after a boostrap).
I managed to do it after many attempts (many dependencies (ant,cmake,g++,python,autoconf,libtool...) we should describe in the README, needed to configure the tools to work properly with our proxy...).
I had to remove the build directory after each encountered error because there were some errors at the next run saying that some directories were already existing.
The clean and cleanup_distribution targets didn't clean up these directories.

I then tried to configure/make/make install the generated package on this UBUNTU 18.04LTS using mariadb as DB.
I didn't manage to find a way to run `configure `so that the DB server would be compiled later on.

Here is the configure command I used (sudo to have mariadb connection OK):
`sudo ../configure --prefix=/home/reynald/install/tango --without-zlib --enable-mariadb --with-mysql-admin=root --with-mariadbclient-include="-I/usr/include" --with-mariadbclient-lib="/usr/lib/x86_64-linux-gnu  -lmariadbclient -lpthread -lz -lm -ldl" --enable-dbcreate --enable-dbserver=yes`

I ended up with something like that at the end of the configure execution:

```
Configuration (tango):

	Source code location:   ..
	Version:                9.3.2
	Compiler:               gcc,g++

	OMNIORB PATH:           /usr/local
	OMNIORB VERSION:        

	ZMQ PATH:               /usr
	ZMQ VERSION:            4.2.5

	JAVA PATH:              
	JAVA VERSION:           10.0.1

	MARIADB CLIENT LIB:  -L/usr/lib/x86_64-linux-gnu  -lmariadbclient -lpthread -lz -lm -ldl 
	MARIADB VERSION:     10.1.29-MariaDB
	MARIADB CONNECTION:  OK

build:
	libraries:              yes
	java application:       no
	access control server:	no
	database server:        no
	database schema create: yes
```

The omniORB version could not be found, I don't know why. omniORB 4.2 was installed from the Ubuntu packages and I updated PKG_CONFIG_PATH so omniORB4.pc could be found.

I also noticed that even though the tango library code is the one from cppTango 9.3.2, `make install` installs a libtango.so.9.2.5 library and liblog4tango libraries (which should have disappeared in cppTango 9.3).
I understand this is coming from the autotools files which have not been fully updated but this shows that we still have quite some work to be done to install correctly the latest TangoDatabase server and cppTango 9.3 library from the Tango source distribution.

File: distribution.properties (hunk 1 above)
Comment (Ingvord) [reply distribution.properties]: Sure

**Comment by Ingvord** [comment]
@bourtemb , thanks a lot for testing this on Ubuntu!

> many dependencies (ant,cmake,g++,python,autoconf,libtool...) we should describe in the README,

Already done in 093f7dc. 

> The clean and cleanup_distribution targets didn't clean up these directories. 

`ant clean` deletes the whole build directory - so it MUST clean up those directories. `ant cleanup_distribution` basically calls legacy `cleanup_distribution` which prepares distribution for packaging i.e. deletes some temporary files etc AFAIU

>   make install installs a libtango.so.9.2.5 library and liblog4tango libraries (which should have 
>   disappeared in cppTango 9.3).

Good catch! ~Looks like this comes from Make.rules file in cppTango/cppapi. So this is just a naming issue, I mean that actual code base installed is of course 9.3.2. Hope this can be fixed easily.~ See #14 and #15 

> I understand this is coming from the autotools files which have not been fully updated but this shows that  we still have quite some work to be done to install correctly the latest TangoDatabase server and cppTango 9.3 library from the Tango source distribution.

Absolutely! We need to migrate to cmake (which we run anyway now to get generated header files) in the long run

**Comment by bourtemb** [comment]
> > many dependencies (ant,cmake,g++,python,autoconf,libtool...) we should describe in the README,
> 
> Already done in [093f7dc](https://github.com/tango-controls/TangoSourceDistribution/commit/093f7dc6d2790c3a2e837c878e63e54dd3fed179).

Great! Thanks!
> 
> > The clean and cleanup_distribution targets didn't clean up these directories.
> 
> `ant clean` deletes the whole build directory - so it MUST clean up those directories. `ant cleanup_distribution` basically calls legacy `cleanup_distribution` which prepares distribution for packaging i.e. deletes some temporary files etc AFAIU

Right you are! I Confirm. `ant clean` deletes build directory. Sorry for the confusion.

**Comment by bourtemb** [comment]
I actually managed to enable database server compilation with mariaDb, I had to use the trick described in #9 .
So on my Ubuntu 18.04LTS VM, I had to use the following configure command:

`sudo ../configure --prefix=/home/reynald/install/tango --without-zlib --enable-mariadb --with-mysql-admin=root --with-mariadbclient-include="/usr/include/mysql" --with-mariadbclient-lib="/usr/lib/x86_64-linux-gnu" --enable-dbcreate --enable-dbserver=yes`

Then I got:
```
Configuration (tango):

	Source code location:   ..
	Version:                9.3.2
	Compiler:               gcc,g++

	OMNIORB PATH:           /usr/local
	OMNIORB VERSION:        

	ZMQ PATH:               /usr
	ZMQ VERSION:            4.2.5

	JAVA PATH:              
	JAVA VERSION:           10.0.1

	MARIADB CLIENT LIB:  -L/usr/lib/x86_64-linux-gnu -lmariadb
	MARIADB VERSION:     10.1.29-MariaDB
	MARIADB CONNECTION:  OK

build:
	libraries:              yes
	java application:       no
	access control server:	yes
	database server:        yes
	database schema create: yes
```

**Comment by Ingvord** [comment]
@bourtemb , I will copy your comment to wiki because this is very important achievement!

**Comment by andygotz** [comment]
Indeed this is excellent news! Well done Igor and all the testers for their feedback!

**Comment by bourtemb** [comment]
And if you install openjdk-8-jre the java tools installation seem to work as well (java application: yes at the end of configure).
It does not work with openjdk-11-jre.

**Comment by Ingvord** [comment]
@bourtemb , thanks. I have updated the wiki page.

**Pull request merged**